Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/Saml2ExchangeAuthFrameworkValidator.java

```diff
@@ -56,31 +59,35 @@
     /** The Constant invalidUserNameChars. */
     private static final char[] invalidUserNameChars = { ';', ':', '\"', '<', '>', '*', '+', '=', '\\', '|', '?', ',' };
 
-    /**
-     * (non-Javadoc).
+    /*
+     * (non-Javadoc)
      *
-     * @param assertion the assertion
-     * @throws ValidationException the validation exception
-     * @see org.opensaml.saml2.core.validator.AssertionSpecValidator#validate(org.opensaml.saml2.core.Assertion)
+     * @see org.apache.wss4j.dom.validate.SamlAssertionValidator#validateAssertion(org.apache.wss4j.common.saml.
+     * SamlAssertionWrapper)
      */
     @Override
-    public void validate(Assertion assertion) throws ValidationException {
-        super.validate(assertion);
+    protected void validateAssertion(final SamlAssertionWrapper samlAssertion) throws WSSecurityException {
+
+        try {
+            super.validateAssertion(samlAssertion);
+            validateSubject(samlAssertion.getSaml2());
+            validateIssuer(samlAssertion.getSaml2().getIssuer());
+
+        } catch (WSSecurityException e) {
+            LOG.error("Validation Fail {}", e.getLocalizedMessage(), e);
+            throw new WSSecurityException(ErrorCode.FAILURE, "Validation fail");
+        }
 
-        validateSubject(assertion);
-        validateIssuer(assertion.getIssuer());
     }
 
     /*
      * (non-Javadoc)
      *
      * @see org.opensaml.saml2.core.validator.AssertionSpecValidator#validateSubject(org.opensaml.saml2.core.Assertion)
      */
-    @Override
-    protected void validateSubject(Assertion assertion) throws ValidationException {
-        validateSubject(assertion.getSubject());
 
-        super.validateSubject(assertion);
+    protected void validateSubject(final Assertion assertion) throws WSSecurityException {
+        validateSubject(assertion.getSubject());
```

Comment:
Agree w/ @mhpnguyen (but for a different reason).  Function overloading is a fantastic thing, as long as the new function adds value.  In this case, however, this function could be removed entirely without any loss of functionality or readability (in the original call, just send the subject directly).